### PR TITLE
upcoming: [M3-8187] - Add useRegionQuery and cleanup bucket landing page

### PR DESCRIPTION
**Pull request by jaalah-akamai**
## Dependency
- [x] https://github.com/linode/manager/pull/10790

## Description 📝
We need to add a new `useRegionQuery` hook to allow querying for a single region in the details drawer. Additionally, we should clean up the `BucketDetailsDrawer` props and deprecate some queries and functions related to OBJ Legacy and Multi-Cluster.

## Changes  🔄
- Implement `useRegionQuery` hook to get a single region by id
- Clean up BucketDetailsDrawer
   - Added unit tests for legacy code. We'll cover other paths in an E2E since that requires mocking flags/account capabilities 
- Deprecated some OBJ Legacy and Multicluster functions and queries

## Target release date 🗓️
N/A

## Preview 📷
No visual changes

### Prerequisites
- Test flow by toggling off all OBJ flags (multi-cluster / gen2)
- Test flow by toggling them on

### Verification steps
- Bucket details drawer should remain unchanged from prod

## As an Author I have considered 🤔

*Check all that apply*

- [x] 👀 Doing a self review
- [x] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [x] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
- [x] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

File: packages/manager/src/features/ObjectStorage/BucketLanding/BucketLanding.tsx [hunk 1]
```diff
@@ -60,44 +60,41 @@ export const BucketLanding = () => {
...
   const [isLoading, setIsLoading] = React.useState<boolean>(false);
   const [error, setError] = React.useState<APIError[] | undefined>(undefined);
   const [
     bucketDetailDrawerOpen,
     setBucketDetailDrawerOpen,
   ] = React.useState<boolean>(false);
-  const [bucketForDetails, setBucketForDetails] = React.useState<
+  const [selectedBucket, setSelectedBucket] = React.useState<
     ObjectStorageBucket | undefined
   >(undefined);
```
Comment (jaalah-akamai) [inline packages/manager/src/features/ObjectStorage/BucketLanding/BucketLanding.tsx:71]: Combined `setBucketToRemove` and `setBucketForDetails` into a single useState `setSelectedBucket`

File: packages/manager/src/queries/regions/regions.ts [hunk 2]
```diff
@@ -27,13 +27,24 @@ export const regionQueries = createQueryKeys('regions', {
...
 });
 
-export const useRegionsQuery = () =>
+export const useRegionQuery = (regionId: string) => {
+  return useQuery<Region, APIError>({
+    ...regionQueries.region(regionId),
+    enabled: Boolean(regionId),
+  });
+};
+
```
Comment (jaalah-akamai) [inline packages/manager/src/queries/regions/regions.ts:46]: Surprised we never had a query like this yet!

**Review by harsh-akamai — APPROVED** [review]
PR looks good 👍

File: packages/manager/src/features/ObjectStorage/BucketLanding/BucketDetailsDrawer.tsx [hunk 3]
```diff
@@ -23,38 +23,31 @@ import { readableBytes } from 'src/utilities/unitConversions';
...
-  bucketLabel?: string;
-  bucketRegion?: Region;
-  cluster?: string;
-  created?: string;
-  endpointType?: ObjectStorageEndpointTypes;
-  hostname?: string;
-  objectsNumber?: number;
   onClose: () => void;
   open: boolean;
-  size?: null | number;
```
Comment (jaalah-akamai) [inline packages/manager/src/features/ObjectStorage/BucketLanding/BucketDetailsDrawer.tsx:41]: Keeping things simple by passing the whole bucket down

File: packages/manager/src/features/ObjectStorage/BucketLanding/BucketDetailsDrawer.tsx [hunk 4]
```diff
@@ -87,68 +86,66 @@ export const BucketDetailsDrawer = React.memo(
...
             <StyledCopyTooltip sx={{ marginLeft: 4 }} text={hostname} />
           </StyledLinkContainer>
-        ) : null}
-        {formattedCreated || cluster ? (
-          <Divider spacingBottom={16} spacingTop={16} />
-        ) : null}
-        {typeof size === 'number' ? (
+        )}
+        {formattedCreated ||
+          (cluster && <Divider spacingBottom={16} spacingTop={16} />)}
```
Comment (coliu-akamai) [inline packages/manager/src/features/ObjectStorage/BucketLanding/BucketDetailsDrawer.tsx:119]:
```suggestion
        {(formattedCreated || cluster) && (
          <Divider spacingBottom={16} spacingTop={16} />
        )}
```

File: packages/manager/src/features/ObjectStorage/BucketLanding/BucketDetailsDrawer.tsx [hunk 5]
```diff
@@ -87,68 +86,66 @@ export const BucketDetailsDrawer = React.memo(
...
           </Link>
-        ) : null}
-        {typeof size === 'number' || typeof objectsNumber === 'number' ? (
-          <Divider spacingBottom={16} spacingTop={16} />
-        ) : null}
+        )}
+        {typeof size === 'number' ||
+          (typeof objects === 'number' && (
+            <Divider spacingBottom={16} spacingTop={16} />
+          ))}
```
Comment (coliu-akamai) [inline packages/manager/src/features/ObjectStorage/BucketLanding/BucketDetailsDrawer.tsx:138]:
```suggestion
        {(typeof size === 'number' || typeof objects === 'number') && (
          <Divider spacingBottom={16} spacingTop={16} />
        )}
```

File: packages/manager/src/queries/regions/regions.ts [hunk 6]
```diff
@@ -27,13 +27,24 @@ export const regionQueries = createQueryKeys('regions', {
...
     queryFn: getAllRegionsRequest,
     queryKey: null,
   },
 });
 
-export const useRegionsQuery = () =>
+export const useRegionQuery = (regionId: string) => {
+  return useQuery<Region, APIError>({
+    ...regionQueries.region(regionId),
+    enabled: Boolean(regionId),
```
Comment (bnussman-akamai) [inline packages/manager/src/queries/regions/regions.ts:43]: For Gecko: I think we'll want to add the `select` here that modifies the `label` so it is consistent with `useRegionsQuery`. 

```suggestion
    enabled: Boolean(regionId),
    select: (region) => ({
      ...region,
      label: getNewRegionLabel(region),
    }),
```

File: packages/manager/src/queries/regions/regions.ts [hunk 7]
```diff
@@ -27,13 +27,24 @@ export const regionQueries = createQueryKeys('regions', {
...
+  });
+};
+
+export const useRegionsQuery = (transformRegionLabel: boolean = false) =>
```
Comment (bnussman-akamai) [inline packages/manager/src/queries/regions/regions.ts:47]: `transformRegionLabel` looks like it is unused. Can we remove it?

**Review by coliu-akamai — COMMENTED** [review]
Not seeing the dividers on the Bucket Details page in comparison to prod/[this pr](https://github.com/linode/manager/pull/10771). Ty for the additional documentation + tests! 😄 

image

File: packages/manager/src/features/ObjectStorage/BucketLanding/BucketDetailsDrawer.test.tsx [hunk 8]
```diff
@@ -0,0 +1,202 @@
...
+// Hoist query mocks
+const queryMocks = vi.hoisted(() => ({
+  useObjectStorageClusters: vi.fn().mockReturnValue({}),
+  useProfile: vi.fn().mockReturnValue({}),
+  useRegionQuery: vi.fn().mockReturnValue({}),
+  useRegionsQuery: vi.fn().mockReturnValue({}),
+}));
+
+// Mock the queries
+vi.mock('src/queries/profile/profile', async () => {
```
[the remark this answers is not in the document]
Comment (jaalah-akamai) [reply packages/manager/src/features/ObjectStorage/BucketLanding/BucketDetailsDrawer.test.tsx:31]: Using Vitest, you don't have to worry about loading states. You can test the component in all it's states and have more control over the data passed in. The downside it's a bit more fragile than the MSW approach where the that tests things in a more E2E type of way, however you have to account for loading states by using `waitFor` in certain cases. Historically it has caused flakyness for us in CI when running multiple tests. Both approaches are valid and you can choose whichever makes sense.

File: packages/manager/src/features/ObjectStorage/BucketLanding/OMC_BucketLanding.tsx [hunk 9]
```diff
@@ -189,17 +168,15 @@ export const OMC_BucketLanding = () => {
           }}
         </OrderBy>
         {/* If there's more than one Bucket, display the total usage. */}
-        {objectStorageBucketsResponse.buckets.length > 1 ? (
+        {buckets.length > 1 ? (
```
Comment (coliu-akamai) [inline packages/manager/src/features/ObjectStorage/BucketLanding/OMC_BucketLanding.tsx:171]: Similar to some of the cleanup you did in BucketDetailsDrawer.tsx, you could probably change this to `buckets.length > 1 && (...)` and get rid of the `: null` on line 178

**Review by coliu-akamai — APPROVED** [review]
retested flows, things look good + bucket details drawers matches prod now! ✅

**Pull request merged**